config/initializers/rack_attack.rb

```diff
@@ -1,5 +1,82 @@
 # frozen_string_literal: true
 
-Rack::Attack.throttle("requests by ip", limit: 20, period: 2, &:ip)
-Rack::Attack.blocklist_ip(ENV.fetch("BLOCKED_IPS")) if ENV.fetch("BLOCKED_IPS", nil).present?
-Rack::Attack.enabled = Rails.env.production? || ENV.key?("ENABLE_RACK_ATTACK")
+module Rack
+  class Attack
+    Rack::Attack.enabled = Rails.env.production? || ENV["ENABLE_RACK_ATTACK"].present?
+
+    Rack::Attack.blocklist_ip(ENV["BLOCKED_IPS"]) if ENV["BLOCKED_IPS"].present?
+
+    ### Configure Cache ###
+
+    # If you don't want to use Rails.cache (Rack::Attack's default), then
+    # configure it here.
+    #
+    # Note: The store is only used for throttling (not blocklisting and
+    # safelisting). It must implement .increment and .write like
+    # ActiveSupport::Cache::Store
+
+    # Rack::Attack.cache.store = ActiveSupport::Cache::MemoryStore.new
+
+    ### Throttle Spammy Clients ###
+
+    # If any single client IP is making tons of requests, then they're
+    # probably malicious or a poorly-configured scraper. Either way, they
+    # don't deserve to hog all of the app server's CPU. Cut them off!
+    #
+    # Note: If you're serving assets through rack, those requests may be
+    # counted by rack-attack and this throttle may be activated too
+    # quickly. If so, enable the condition to exclude them from tracking.
+
+    # Throttle all requests by IP (60rpm)
+    #
+    # Key: "rack::attack:#{Time.now.to_i/:period}:req/ip:#{req.ip}"
+    throttle("req/ip", limit: 300, period: 5.minutes, &:ip)
```

Comment:
It would be nice to be able to vary this based on whether you are authenticated.

Some apps might have javascript polling each second or similar.
For companies using vpn, a lot of users would come from the same ip.

Reply:
Should we limit this throttling to post?